### ISSUE TITLE
Bug: wrong indent in `else-if`

### ISSUE THREAD

**Issue by otomad** [eslint/eslint#17316]
### Environment

Node version: v19.0.1
npm version: v8.19.2
Local ESLint version: v8.43.0
Global ESLint version: Not found
Operating System: win32 10.0.22621

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```json
"rules": {
	"curly": ["error", "multi"],
	"indent": ["error", "tab"],
}
```
</details>

```js
const foo = true, bar = true, doSomething = () => { };

if (foo)
	if (bar) doSomething();
	else doSomething();
else
	if (bar) doSomething();
	else doSomething();
```

### What did you expect to happen?

The demo code is correct and does not require to fix.

### What actually happened?

The `else` part indents marked as error.

```js
if (foo)
	if (bar) doSomething();
	else doSomething();
else
﹏﹏﹏﹏ if (bar) doSomething();
﹏﹏﹏﹏ else doSomething();
```

Problem content:

```
Expected indentation of 0 tabs but found 1. eslint(indent)
```

After fix all auto-fixable problems

```js
if (foo)
	if (bar) doSomething();
	else doSomething();
else
if (bar) doSomething();
else doSomething();
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/js-efzdhw?file=index.js&view=editor

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Rec0iL99** [comment]
Hi @otomad, thanks for the issue. I was able to reproduce the error. Removing the `curly` rule also produced the same results. The issue seems to go away when using block statements.

[playground](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IFwiaW5kZW50XCI6IFtcImVycm9yXCIsIFwidGFiXCJdICovXG5cbi8vIHVzaW5nIGJsb2NrIHN0YXRlbWVudHNcblxuaWYgKGZvbykge1xuXHRpZiAoYmFyKSBkb1NvbWV0aGluZygpO1xuXHRlbHNlIGRvU29tZXRoaW5nKCk7XG59XG5lbHNlIHtcblx0aWYgKGJhcikgZG9Tb21ldGhpbmcoKTtcblx0ZWxzZSBkb1NvbWV0aGluZygpO1xufVxuXG4vLyBOT1QgdXNpbmcgYmxvY2sgc3RhdGVtZW50c1xuXG5pZiAoZm9vKVxuXHRpZiAoYmFyKSBkb1NvbWV0aGluZygpO1xuXHRlbHNlIGRvU29tZXRoaW5nKCk7XG5lbHNlXG4gICAgaWYgKGJhcikgZG9Tb21ldGhpbmcoKTtcbiAgICBlbHNlIGRvU29tZXRoaW5nKCk7XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImVzNiI6dHJ1ZSwiYnJvd3NlciI6dHJ1ZX19fQ==)

**Comment by mdjermanovic** [comment]
Looks like a bug to me. I'm working on this.

**Comment by GermanBluefox** [comment]
Everything fine. But now how to deactivate this rule only for this particular case?
```
const foo = true, bar = true, doSomething = () => { };

if (foo) {
    doSomething(true);
} else
if (bar) {
   doSomething(false);
}
```
In this case I have `indent` error on the line with `if (bar) {`

**Comment by darksabrefr** [comment]
I confirm what @GermanBluefox says, the fix have broken things, maybe this issue should be re-opened?

**Comment by Rec0iL99** [comment]
@GermanBluefox @darksabrefr We lost track of this since we generally don't closely track closed issues. Please open a new issue with proper playground examples explaining the bug. Thanks.
